connector/kafka-0-10-sql/src/main/scala/org/apache/spark/sql/kafka010/consumer/FetchedDataPool.scala

```diff
@@ -139,7 +139,7 @@ private[consumer] class FetchedDataPool(
   }
 
   def reset(): Unit = synchronized {
-    scheduled.foreach(_.cancel(true))
+    scheduled.foreach(_.cancel(false))
```

Comment:
```
      val future = executorService.scheduleAtFixedRate(() => {
        Utils.tryLogNonFatalError(removeIdleFetchedData())
      }, 0, evictorThreadRunIntervalMillis, TimeUnit.MILLISECONDS)
      Some(future)
```
There is no method called in the task that will throw `InterruptdEException`.

Reply:
Why change this though? if it were interruptible, we'd want to interrupt for sure.
What is the issue if you try to interrupt something and it doesn't do anything?

Reply:
These tasks can't response interrupt even if we want to interrupt them. `cancel(true)` causes thread interrupt, but it will not affect these tasks that can't response interrupt. In addition, interrupting operations have extra overhead.

Reply:
If true, then this change doesn't do anything, right? I wonder what problem this change is solving.

Reply:
This PR want avoid the overhead of the invalid thread interrupt.

Reply:
Is there overhead? it just sets the interrupted status of the thread.
I'm worried that, if the code changed in some way that would make it interruptible, then we lose the ability to interrupt on cancel.

Reply:
One hand, I want avoid the overhead of the invalid thread interruption, even if the overhead is not big enough.
On the other hand, I suggest the reasonable use of interrupts: use interrupt mechanisms only when needed to avoid excessive dependence.